[Dev] App crash

The application crashed when deleting the user account.

```
************ CAUSE OF ERROR ************

java.lang.RuntimeException: Failure delivering result ResultInfo{who=null, request=101, result=-1, data=Intent { (has extras) }} to activity {com.nextcloud.android.beta/com.owncloud.android.ui.activity.FileDisplayActivity}: java.util.NoSuchElementException: No value present
	at android.app.ActivityThread.deliverResults(ActivityThread.java:5471)
	at android.app.ActivityThread.handleSendResult(ActivityThread.java:5512)
	at android.app.servertransaction.ActivityResultItem.execute(ActivityResultItem.java:51)
	at android.app.servertransaction.TransactionExecutor.executeCallbacks(TransactionExecutor.java:149)
	at android.app.servertransaction.TransactionExecutor.execute(TransactionExecutor.java:103)
	at android.app.ActivityThread$H.handleMessage(ActivityThread.java:2386)
	at android.os.Handler.dispatchMessage(Handler.java:107)
	at android.os.Looper.loop(Looper.java:213)
	at android.app.ActivityThread.main(ActivityThread.java:8178)
	at java.lang.reflect.Method.invoke(Native Method)
	at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:513)
	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:1101)
Caused by: java.util.NoSuchElementException: No value present
	at com.nextcloud.java.util.Optional.get(Optional.java:142)
	at com.nextcloud.client.mixins.SessionMixin.setAccount(SessionMixin.kt:68)
	at com.owncloud.android.ui.activity.BaseActivity.setAccount(BaseActivity.java:131)
	at com.owncloud.android.ui.activity.DrawerActivity.onActivityResult(DrawerActivity.java:1005)
	at com.owncloud.android.ui.activity.FileDisplayActivity.onActivityResult(FileDisplayActivity.java:941)
	at android.app.Activity.dispatchActivityResult(Activity.java:8413)
	at android.app.ActivityThread.deliverResults(ActivityThread.java:5464)
	... 11 more

************ APP INFORMATION ************
ID: com.nextcloud.android.beta
Version: 20220112
Build flavor: versionDev
```

--- a/src/main/java/com/nextcloud/client/mixins/SessionMixin.kt
+++ b/src/main/java/com/nextcloud/client/mixins/SessionMixin.kt
@@ -91,6 +91,7 @@ class SessionMixin constructor(
         val newAccount = accountManager.currentAccount
         if (newAccount == null) {
             // no account available: force account creation
+            currentAccount = null
             startAccountCreation()
         } else {
             currentAccount = newAccount

